vcd/resource_vcd_lb_server_pool.go

```diff
@@ -0,0 +1,344 @@
+	if err != nil {
+		return fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	}
+
+	readLBPool, err := edgeGateway.ReadLBServerPool(&types.LBPool{ID: d.Id()})
+	if err != nil {
+		d.SetId("")
+		return fmt.Errorf("unable to find load balancer server pool with ID %s: %s", d.Id(), err)
+	}
+
+	return setLBPoolData(d, readLBPool)
+}
+
+func resourceVcdLBServerPoolUpdate(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	vcdClient.lockParentEdgeGtw(d)
+	defer vcdClient.unLockParentEdgeGtw(d)
+
+	edgeGateway, err := vcdClient.GetEdgeGatewayFromResource(d, "edge_gateway")
+	if err != nil {
+		return fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	}
+
+	updateLBPoolConfig, err := getLBPoolType(d)
+	if err != nil {
+		return fmt.Errorf("could not expand load balancer server pool for update: %s", err)
+	}
+
+	updatedLBPool, err := edgeGateway.UpdateLBServerPool(updateLBPoolConfig)
+	if err != nil {
+		return fmt.Errorf("unable to update load balancer server pool with ID %s: %s", d.Id(), err)
+	}
+
+	return setLBPoolData(d, updatedLBPool)
+}
+
+func resourceVcdLBServerPoolDelete(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	vcdClient.lockParentEdgeGtw(d)
+	defer vcdClient.unLockParentEdgeGtw(d)
+
+	edgeGateway, err := vcdClient.GetEdgeGatewayFromResource(d, "edge_gateway")
+	if err != nil {
+		return fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	}
+
+	err = edgeGateway.DeleteLBServerPool(&types.LBPool{ID: d.Id()})
+	if err != nil {
+		return fmt.Errorf("error deleting load balancer server pool: %s", err)
+	}
+
+	d.SetId("")
+	return nil
+}
+
+func resourceVcdLBServerPoolImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	resourceURI := strings.Split(d.Id(), ".")
+	if len(resourceURI) != 4 {
+		return nil, fmt.Errorf("resource name must be specified as org.VDC.edge-gw.lb-server-pool")
+	}
+	orgName, vdcName, edgeName, poolName := resourceURI[0], resourceURI[1], resourceURI[2], resourceURI[3]
+
+	vcdClient := meta.(*VCDClient)
+	edgeGateway, err := vcdClient.GetEdgeGateway(orgName, vdcName, edgeName)
+	if err != nil {
+		return nil, fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	}
+
+	readLBPool, err := edgeGateway.ReadLBServerPool(&types.LBPool{Name: poolName})
+	if err != nil {
+		return []*schema.ResourceData{}, fmt.Errorf("unable to find load balancer server pool with name %s: %s", d.Id(), err)
+	}
+
+	d.Set("org", orgName)
+	d.Set("vdc", vdcName)
+	d.Set("edge_gateway", edgeName)
+	d.Set("name", poolName)
+
+	d.SetId(readLBPool.ID)
+	return []*schema.ResourceData{d}, nil
+}
+
+// getLBPoolType converts schema.ResourceData to *types.LBPool and is useful
+// for creating API requests
+func getLBPoolType(d *schema.ResourceData) (*types.LBPool, error) {
+	lbPool := &types.LBPool{
+		Name:                d.Get("name").(string),
+		Description:         d.Get("description").(string),
```

Comment:
It won't panic if isn't defined?

Reply:
It won't. Adjusted test to cover empty description as well.